compiler/rustc_hir_analysis/src/outlives/mod.rs

```diff
@@ -46,6 +47,24 @@ fn inferred_outlives_of(tcx: TyCtxt<'_>, item_def_id: LocalDefId) -> &[(ty::Clau
                 &[]
             }
         }
+        DefKind::Impl { of_trait: false }
+            if let hir::ItemKind::Impl(item) = tcx.hir().expect_item(item_def_id).kind
+                && let hir::Impl { self_ty, .. } = &item
+                && let hir::TyKind::TraitObject(
+                    _,
+                    hir::Lifetime { res: hir::LifetimeName::ImplicitObjectLifetimeDefault, .. },
+                    _,
+                ) = self_ty.kind =>
+        {
+            // `impl dyn Trait {}` has an implicit `dyn Trait: 'static` bound. We add it here for
+            // more context in errors. With this we point at `impl dyn Trait` and not to it's
+            // associated functions, deduplicating errors to a single one per `impl dyn Trait` instead
+            // of one for it plus one for each associated function.
+            let ty = tcx.type_of(item_def_id).instantiate_identity();
```

Comment:
Also, this should at least be asserting that the implicit object outlives is actually static:

```
let ty::Dynamic(_, outlives, _) = *ty.kind() else {
    bug!("expected dyn self type");
};
assert_eq!(outlives, tcx.regions.re_static);
```